src/quality_result_gui/quality_error_manager_settings.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright (C) 2023 National Land Survey of Finland
+#  (https://www.maanmittauslaitos.fi/en).
+#
+#
+#  This file is part of quality-result-gui.
+#
+#  quality-result-gui is free software: you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License as published
+#  by the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  quality-result-gui is distributed in the hope that it will be
+#  useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+#  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with quality-result-gui. If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Dict, Optional
+
+from qgis.core import QgsVectorLayer
+from qgis.PyQt.QtCore import QObject
+
+
+class QualityResultManagerSettings(QObject):
+    """
+    Singleton for storing common properties.
+    """
+
+    _instance: Optional["QualityResultManagerSettings"] = None
+    layer_mapping: Dict[str, QgsVectorLayer] = {}
```

Comment:
I think we need to do mapping as Dict[str, str] where instead of QgsVectorLayer instance we use the layer id. Otherwise we might get the error "RuntimeError: wrapped C/C++ object of type QgsVectorLayer has been deleted" if the underlying vector layer is deleted by the user.

The handling of deleted layer needs to be also done somewhere. Maybe layer mapping could be converted to dataclass and then add methods like `get_layer_name(feature_type: str) -> str` and `get_field_alias(feature_type: str, attribute: str) -> str`?